### PR TITLE
Init model

### PR DESCRIPTION
**Pull request by truongnv0211**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  has_many :lessons, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+
+  validates :email, format: { with: VALID_EMAIL_REGEX },
+                    presence: true, length: { maximum: 255 },
+                    uniqueness: { case_sensitive: false }
```
Comment (vjnhat) [inline app/models/user.rb]: remove space after { and before }

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Category < ActiveRecord::Base
+  has_many :words, dependent: :destroy
+
+  validates :title, presence: true
+
```
Comment (nguyenduyta) [inline app/models/category.rb]: remove line blank

File: app/models/lesson.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Lesson < ActiveRecord::Base
+  has_many :lesson_words, dependent: :destroy
+  belongs_to :user
+
```
Comment (nguyenduyta) [inline app/models/lesson.rb]: remove line blank

File: app/models/lesson_word.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class LessonWord < ActiveRecord::Base
+  belongs_to :lesson
+  belongs_to :words
+
```
Comment (nguyenduyta) [inline app/models/lesson_word.rb]: remove line blank

File: app/models/relationship.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: "User"
+  belongs_to :followed, class_name: "User"
+
+  validates :follower_id, presence: true
+  validates :followed_id, presence: true
+
```
Comment (nguyenduyta) [inline app/models/relationship.rb]: remove line blank

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class User < ActiveRecord::Base
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  has_many :lessons, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+
+  validates :email, format: { with: VALID_EMAIL_REGEX },
```
Comment (nguyenduyta) [inline app/models/user.rb]: Not space after and before { }

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+                    presence: true, length: { maximum: 255 },
+                    uniqueness: { case_sensitive: false }
+  validates :username, presence: true, length: { maximum: 50 }
+
```
Comment (nguyenduyta) [inline app/models/user.rb]: remove line blank

File: app/models/word.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class Word < ActiveRecord::Base
+  belongs_to :category
+  has_many :lesson_words, dependent: :destroy
+
+  validates :jp_word, presence: true
+  validates :correct_answer, presence: true, numericality: { only_integer: true }
+
```
Comment (nguyenduyta) [inline app/models/word.rb]: nt

File: config/database.yml.example [hunk 9]
```diff
@@ -1,25 +0,0 @@
-# SQLite version 3.x
```
Comment (nguyenduyta) [inline config/database.yml.example]: Not delete this file

**Comment by vjnhat** [comment]
3 commit ????

**Pull request closed without merging**